packages/eslint-plugin/src/rules/shorthand-property-sorting/index.ts

```diff
@@ -4,123 +4,419 @@ import {
+    return [];
+  }
+
+  for (const property of objectExpression.properties) {
+    if (property.type === 'SpreadElement') {
+      continue;
+    }
+
+    if (property.value.type !== 'ObjectExpression') {
+      continue;
+    }
+
+    if (property.key.type !== 'Literal' && property.key.type !== 'Identifier') {
+      continue;
+    }
+
+    // Unconditionally traverse through the whole cssMap object, for simplicity.
+    //
+    // Not very performant and can give false positives, but considering that
+    // the cssMap key can be dynamic, we at least avoid any false negatives.
+    //
+    // (https://compiledcssinjs.com/docs/api-cssmap#dynamic-declarations)
+    properties.push(...getObjectCSSProperties(context, property.value));
+  }
+
+  return properties;
+};
+
+const parseStyled = (context: Rule.RuleContext, node: CallExpression): PropertyArray => {
+  // Handle styled
+  const { references } = context.sourceCode.getScope(node);
+  if (!isStyled(node.callee as Rule.Node, references)) {
+    return [];
+  }
+
+  const properties: PropertyArray = [];
+
+  if (node.callee.type === 'CallExpression' && node.callee.arguments.length) {
+    // This is the BaseComponent in styled(BaseComponent)({ ... })
+    const baseComponentName = node.callee.arguments[0];
+    if (baseComponentName.type === 'Identifier') {
+      // Assume that the styled function call and the definition of BaseComponent
+      // are both defined in the same scope, presumably global scope.
+      //
+      // (Not sure how many people don't use it this way)
+      const baseComponentDefinition = references.find(
+        (ref) => ref.identifier === baseComponentName
+      )?.resolved;
+
+      if (
+        baseComponentDefinition &&
+        baseComponentDefinition.defs.length &&
+        baseComponentDefinition.defs[0].node.type === 'VariableDeclarator' &&
+        baseComponentDefinition.defs[0].node.init.type === 'CallExpression'
+      ) {
+        const baseComponent = baseComponentDefinition.defs[0].node.init as CallExpression;
+        properties.push(...parseStyled(context, baseComponent));
+      }
+    }
+  }
+
+  for (const argument of node.arguments) {
+    if (argument.type === 'ObjectExpression') {
+      properties.push(...getObjectCSSProperties(context, argument));
+    } else if (argument.type === 'Identifier' || argument.type === 'LogicalExpression') {
+      // Handles this scenario:
+      //
+      //     const paddingStyles = css({ padding: '...' });
+      //     const Component = styled.div(
+      //         paddingStyles,
+      //         { margin: '...' },
+      //     )
+
+      // For arguments of `styled` function calls, we can re-use some of the same
+      // logic that we use to parse elements in the css prop array:
+      //
+      //            vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
+      // <div css={[baseStyles, someCondition && someMoreStyles]}>Hello</div>
+      properties.push(...parseCssArrayElement(context, argument));
```

Comment:
I may not exactly be able to follow everything from a PR review, the comments make it feasible that I could follow it running the tests locally and digging into what I need, thank you for them!

Reply:
ah i'm happy to add more comments to other parts of the file to help you with review - let me know if any part doesn't immediately make sense